ls1intum/Artemis
# Allow instructors to cancel assessments

### Checklist

### Motivation and Context
This PR solves issue #1150. Right now, instructors are not able to release the lock for assessments if a tutor has it. With this PR, instructors will be able to cancel an assignment (e.g. if a tutor does not complete it or another tutor should do it). This can be done from the submissions and assessments page via the new button "Cancel assessment" or from the assessment detail page via the existing button "Cancel".

### Description
1. Add button "Cancel assessment" to submissions and assessments page (for TextExercise, FileUploadExercise, ModelingExercise) which is only available to instructors and only shown if an assessment exists and it is not completed yet
2. In `/cancel-assessment` endpoints in backend, allow instructor to cancel an assessment (despite not being the assessor themselves)
3. Adapt existing button "Cancel" on assessment detail page to new permission logic (should not be disabled if user is instructor and not completed yet)

### Steps for Testing
1. Log in to Artemis
2. Navigate to Overview
3. Click on a Course for which the user is an instructor and that has a TextExercise, FileUploadExercise, ModelingExercise
4. Click on all those exercises
5. Click on the instructor action button "Submissions"
6. Verify that the button "Cancel assessment" is shown (together with "Open assessment") if there is an assessment
7. Click on "Cancel assessment" and see that the result is removed and the reviewer is unset

### Screenshots
![OnPaste 20200107-161811](https://user-images.githubusercontent.com/7109225/71906228-f5991980-3169-11ea-9043-d0c1dd9fc7a3.png)
![OnPaste 20200107-161843](https://user-images.githubusercontent.com/7109225/71906230-f5991980-3169-11ea-9230-de8a52dcc45b.png)
![OnPaste 20200107-162241](https://user-images.githubusercontent.com/7109225/71906231-f5991980-3169-11ea-97a7-7a8920b4eb8f.png)
